cli/template/base/src/env.mjs

```diff
@@ -32,16 +32,21 @@ const processEnv = {
 // --------------------------
 
 const merged = server.merge(client);
-/** @type z.infer<merged>
- *  @ts-ignore - can't type this properly in jsdoc */
-let env = process.env;
+
+/** @typedef {z.input<typeof merged>} MergedInput */
+/** @typedef {z.infer<typeof merged>} MergedOutput */
+/** @typedef {z.SafeParseReturnType<MergedInput, MergedOutput>} MergedSafeParseReturn */
+
+let env = /** @type {MergedOutput} */ (process.env);
 
 if (!!process.env.SKIP_ENV_VALIDATION == false) {
   const isServer = typeof window === "undefined";
 
-  const parsed = isServer
-    ? merged.safeParse(processEnv) // on server we can validate all env vars
-    : client.safeParse(processEnv); // on client we can only validate the ones that are exposed
+  const parsed = /** @type {MergedSafeParseReturn} */ (
+    isServer
+      ? merged.safeParse(processEnv) // on server we can validate all env vars
+      : client.safeParse(processEnv) // on client we can only validate the ones that are exposed
+  );
```

Comment:
do you need the type in the middle like that? doesn't this work?

```suggestion
  /** @type {MergedSafeParseReturn} */
  const parsed = isServer
    ? merged.safeParse(processEnv) // on server we can validate all env vars
    : client.safeParse(processEnv); // on client we can only validate the ones that are exposed
```

Reply:
It doesn't work like that, we need this syntax for the type casting: https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html#casts

Reply:
I see

Reply:
![Screenshot from 2023-02-09 17-43-15](https://user-images.githubusercontent.com/25031305/217935012-52914021-b58b-4267-b9ee-98b39a2f7d3e.png)

The type casting works kinda like the `as` keyword in TS